lms/static/scripts/frontend_apps/components/dashboard/StudentsActivity.tsx

```diff
@@ -20,10 +21,11 @@ export default function StudentsActivity() {
   const { routes } = dashboard;
   const { assignmentId } = useParams<{ assignmentId: string }>();
   const assignment = useAPIFetch<Assignment>(
-    routes.assignment.replace(':assignment_id', assignmentId),
+    replaceURLParams(routes.assignment, { assignment_id: assignmentId }).url,
   );
   const students = useAPIFetch<StudentStats[]>(
-    routes.assignment_stats.replace(':assignment_id', assignmentId),
+    replaceURLParams(routes.assignment_stats, { assignment_id: assignmentId })
+      .url,
```

Comment:
I'm ignoring the `unusedParams` for now. In parallel I'm working on wrapping this component in a `DataLoader` and centralizing the error handling and loading state there. I will verify `unusedParams` there.